runtime/src/eval/system.rs

```diff
@@ -448,8 +448,8 @@ pub fn create<H: Handler>(runtime: &mut Runtime, is_create2: bool, handler: &mut
 	};
 
 	match handler.create(runtime.context.address, scheme, value, code, None) {
-		Capture::Exit((reason, address, return_data)) => {
-			match super::finish_create(runtime, reason, address, return_data) {
```

Comment:
I don't think we can remove the address return here because `finish_create` pushes the address onto the EVM stack which I think is required for the `CREATE` instruction.

Reply:
Ah, I see. The `Capture::Exit` case used to always have `address: None` because it was only returned when the call to `create_inner` encountered an error and therefore no contract was created. The happy path is returning `Capture::Trap` and `finish_create` ends up being called later with the created address as part of the EVM call stack handling logic. Maybe it is worth adding a comment that the `Capture::Exit` case only happens if there was an error. Maybe also worth adding `debug_assert!(!reason.is_succeed())` to make sure this assumption is upheld in the future. 

As for removing unused code, I fully agree with doing that to make maintaining the code easier going forward! I just wanted to make sure the code path really was unused.

Reply:
I agree, thanks. 
In order to get closer to proving correctness, I implemented 100% passing of tests from [ethereum/tests](https://github.com/ethereum/tests) and [ethereum/execution-spec-tests](https://github.com/ethereum/execution-spec-tests). And this is a big step forward. For greater persuasiveness, it is necessary to cover the entire code with tests. However, this task will take, in my opinion, a lot of time, since the EVM logic is really complex. This does not mean that it should not be done - it is just a lower priority.